merlin-worker/src/main/java/gov/nasa/jpl/aerie/merlin/worker/ListenSimulationCapability.java

```diff
@@ -0,0 +1,60 @@
+package gov.nasa.jpl.aerie.merlin.worker;
+
+import com.impossibl.postgres.api.jdbc.PGConnection;
+import com.impossibl.postgres.api.jdbc.PGNotificationListener;
+import gov.nasa.jpl.aerie.merlin.server.remotes.postgres.DatabaseException;
+import gov.nasa.jpl.aerie.merlin.worker.postgres.ListenSimulationStatusAction;
+import gov.nasa.jpl.aerie.merlin.worker.postgres.PostgresSimulationNotificationPayload;
+
+import javax.json.Json;
+import javax.sql.DataSource;
+import java.io.StringReader;
+import java.sql.SQLException;
+import java.util.concurrent.BlockingQueue;
+
+import static gov.nasa.jpl.aerie.merlin.worker.postgres.PostgresNotificationJsonParsers.postgresSimulationNotificationP;
+
+public class ListenSimulationCapability {
+  private final DataSource dataSource;
+  private final BlockingQueue<PostgresSimulationNotificationPayload> notificationQueue;
+
+  public ListenSimulationCapability(
+      final DataSource dataSource,
+      final BlockingQueue<PostgresSimulationNotificationPayload> notificationQueue) {
+    this.dataSource = dataSource;
+    this.notificationQueue = notificationQueue;
+  }
+
+  public void registerListener() throws SQLException {
+    PGConnection connection = this.dataSource.getConnection().unwrap(PGConnection.class);
```

Comment:
🔶 I'm intrigued by this unwrap - I wonder if it's possible to avoid it. In particular it seems like `DataSource` could be more specific... I'll let you know if I find an alternate way of expressing this.

Reply:
From a cursory look at the types, it seems like your approach is the expected approach 👍  I'll resolve this comment now.

Reply:
Bummer. I was hoping you would find something I hadn't found :) . I agree, it feels dirty/wrong.